File: 1662_CheckIfTwoStringArraysAreEquivalent.cpp
// Name: Tanisha Garg
// Date: 25-10-2022

class Solution {
public:
    bool arrayStringsAreEqual(vector<string>& word1, vector<string>& word2) {
        // word pointers
        int i = 0, j = 0; 
        // char pointers
        int m = 0, n = 0; 
        // iteration over both string arrays simultaneously
        while (i < word1.size() and j < word2.size()) {  
            // if letter is not equal return false
            if (word1[i][m++] != word2[j][n++])
                return false;
            
            if (m >= word1[i].size())
                i++, m = 0;
            
            if (n >= word2[j].size())
                j++, n = 0;
        }
        // return true if both array size is same
        return i == word1.size() and j == word2.size();
    }
};

File: PYTHON/1662_CheckIfTwoStringArraysAreEquivalent.py
# Name : Simran Kukreja
# Date : 26th Oct 2022

class Solution(object):
    def arrayStringsAreEqual(self, word1, word2):
        """
        :type word1: List[str]
        :type word2: List[str]
        :rtype: bool
        """
        str1 = ""
        str2 = ""
        
        # Create a single string of all elements in array 1 
        for i in word1:
            str1 += i
            
        # similarly in array 2
        for i in word2:
            str2 += i
            
        # check if both are equal or not
        return (str1 == str2)
